### DISCUSSION TITLE
Include Sedutil

### DISCUSSION THREAD

**Discussion by jose-pr** [zbm-dev/zfsbootmenu#332] [Feature Requests]
Does ZBM currently include sedutil?
 If not i would like to request to allow unlocking boot drives. A hook like the one currently in contrib for luks would prompt for password if it cant find any bootable disks and the disks it finds are sed encrypted. 
At least including sedutil in the inage would allow manually unlocking the disks or using a script to unlock them.

**Comment by ahesford** [comment]
Sedutil is not included in the default ZBM images and seems too specialized for general inclusion. You should build a custom image that pulls in the tools you need to unlock your pool.

PRs for contrib scripts that perform advanced unlocking would be welcome.

**Reply by jose-pr** [reply]
@ahesford I am trying to see how to build the custom image, should I be adding sedutil to xbps-install in https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/docker/Dockerfile , after that would the it generated images include the tools/package added?

**Reply by ahesford** [reply]
While you can add `sedutil` to the container image, there should be no need. The containerized build process has been substantially reworked recently. A worked-through example is available [in the wiki](https://github.com/zbm-dev/zfsbootmenu/wiki/Building-in-Containers). You can add your necessary configuration and scripts in `/etc/zfsbootmenu` in the same way as I do in that example, and then add

```sh
echo "BUILD_ARGS+=( -p sedutil )" >> /etc/zfsbootmenu/zbm-builder.conf
```

to tell the builder to pull `sedutil` into the running container.
